metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1195,26 +1197,13 @@ def get_tables_for_dataset(
 
                     table_items[table.table_id] = table
 
-                if str(table_identifier).startswith(
-                    self.config.temp_table_dataset_prefix
-                ):
-                    logger.debug(f"Dropping temporary table {table_identifier.table}")
-                    self.report.report_dropped(table_identifier.raw_table_name())
-                    continue
-
-                table_count_in_this_batch += 1
-
                 if (
-                    table_count_in_this_batch
-                    % self.config.number_of_datasets_process_in_batch
+                    len(table_items) % self.config.number_of_datasets_process_in_batch
```

Comment:
I think this can also become a `==` because we clear `table_items()` on querying